skore/src/skore/ui/project_routes.py

```diff
@@ -54,145 +35,17 @@ class SerializableProject:
-                "value": json.loads(item.plot_bytes.decode("utf-8")),
-            }
-        ],
-        "sections": [
-            {
-                "title": "Model",
-                "icon": "icon-square-cursor",
-                "items": [
-                    {
-                        "name": "Estimator parameters",
-                        "description": "Core model configuration used for training",
-                        "value": estimator_params_as_str,
-                    },
-                    {
-                        "name": "Cross-validation parameters",
-                        "description": "Controls how data is split and validated",
-                        "value": cv_params_as_str,
-                    },
-                ],
-            }
-        ],
-    }
-
-
-def __pandas_dataframe_as_serializable(df: pandas.DataFrame):
-    return df.fillna("NaN").to_dict(orient="tight")
-
-
 def __item_as_serializable(name: str, item: Item) -> SerializableItem:
-    if isinstance(item, PrimitiveItem):
-        value = item.primitive
-        media_type = "text/markdown"
-    elif isinstance(item, NumpyArrayItem):
-        value = item.array.tolist()
-        media_type = "text/markdown"
-    elif isinstance(item, PandasDataFrameItem):
-        value = __pandas_dataframe_as_serializable(item.dataframe)
-        media_type = "application/vnd.dataframe"
-    elif isinstance(item, PandasSeriesItem):
-        value = item.series.fillna("NaN").to_list()
-        media_type = "text/markdown"
-    elif isinstance(item, PolarsDataFrameItem):
-        value = __pandas_dataframe_as_serializable(item.dataframe.to_pandas())
-        media_type = "application/vnd.dataframe"
-    elif isinstance(item, PolarsSeriesItem):
-        value = item.series.to_list()
-        media_type = "text/markdown"
-    elif isinstance(item, SklearnBaseEstimatorItem):
-        value = item.estimator_html_repr
-        media_type = "application/vnd.sklearn.estimator+html"
-    elif isinstance(item, MediaItem):
-        if "text" in item.media_type:
-            value = item.media_bytes.decode(encoding=item.media_encoding)
-            media_type = f"{item.media_type}"
-        else:
-            value = base64.b64encode(item.media_bytes).decode()
-            media_type = f"{item.media_type};base64"
-    elif isinstance(item, CrossValidationItem):
-        value = __cross_validation_item_as_serializable(item)
-        media_type = "application/vnd.skore.cross_validation+json"
-    else:
-        raise ValueError(f"Item {item} is not a known item type.")
-
+    try:
+        d = item.as_serializable_dict()
+    except Exception as e:
+        raise ValueError(f"Item {item} is not a known item type.") from e
     return SerializableItem(
         name=name,
-        media_type=media_type,
-        value=value,
-        updated_at=item.updated_at,
-        created_at=item.created_at,
+        media_type=d["media_type"],
+        value=d["value"],
```

Comment:
Note that you're assuming here that `as_serializable_dict` returns a dict with the keys `media_type` and `value`.

Reply:
Used get to ensure no `KeyError`.